### DISCUSSION TITLE
How to modularise ent?

### DISCUSSION THREAD

**Discussion by ramonmedeiros** [ent/ent#2992] [General]
Hi guys,

The codebase that I work with ent is growing and I would like to modularise a little bit, with the goal to separated some entities in modules, so I can do separated migrations and in case some of them change, does not affect other entities.

So, let's say I have 3 or 4 ent modules in my codebase. How these cases work:

1) Let's say entity A(module 1) has an edge with entity B (module2). Is it possible to keep them connected? use hooks and stuff?

2) Will my app have one ent client or multiple?

These are the cases that come to my mind, feel free to bring more cases

**Comment by a8m** [comment]
Hey @ramonmedeiros, I hope you don't mind, but I converted https://github.com/ent/ent/issues/2991 to this discussion.  

> so I can do separated migrations and in case some of them change, does not affect other entities.

Can you elaborate on this? Do all entities reside on the same database/schema? What do you mean by separated migrations? Do you use versioned or declarative migrations? Note that Ent changes only the ent/schemas that were changed.
